src/Datadog.Trace/DiagnosticListeners/DiagnosticObserver.cs

```diff
@@ -41,6 +41,10 @@ public virtual IDisposable SubscribeIfMatch(DiagnosticListener diagnosticListene
 
         void IObserver<KeyValuePair<string, object>>.OnNext(KeyValuePair<string, object> value)
         {
+#if DEBUG
+            // In debug mode we allow exceptions to be catch in the test suite
+            OnNext(value.Key, value.Value);
```

Comment:
Missing a `return`. Either way, I think it's better to add a `#if DEBUG throw;` in the catch block instead (so we keep the log in debug mode)

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1061/commits/b948ceba50feaa8382129fe7dc7a06a1b5034835